atomic_reactor/plugins/pre_check_and_set_platforms.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright (c) 2018 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+
+Query the koji build target, if any, to find the enabled architectures. Remove any excluded
+architectures, and return the resulting list.
+
+build_orchestrate_build will prefer this list of architectures over the platforms supplied by
+USER_PARAMS, which is necessary to allow autobuilds to build on the correct architectures
+when koji build tags change.
+"""
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import get_platform_limits
+from atomic_reactor.plugins.pre_reactor_config import get_koji_session
+from atomic_reactor.constants import PLUGIN_CHECK_AND_SET_PLATFORMS_KEY
+
+
+class CheckAndSetPlatformsPlugin(PreBuildPlugin):
+    key = PLUGIN_CHECK_AND_SET_PLATFORMS_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, target, hub=None, root=None, proxy=None,
+                 koji_ssl_certs_dir=None):
+
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param target: string, koji target to use as a source
+        :param hub: string, koji hub (xmlrpc)
```

Comment:
I'm wondering if we should not bother with providing these environment plugin parameters. This plugin is meant to run starting on arrangement 6 which requires reactor config map.

Reply:
Fixed.